web/src/components/modals/fork-pep.tsx

```diff
@@ -22,6 +23,41 @@ interface ForkProjectInputs {
   is_private: boolean;
 }
 
+type CombinedErrorMessageProps = {
+  errors: FieldErrors<ForkProjectInputs>;
+};
+
+const CombinedErrorMessage = (props: CombinedErrorMessageProps) => {
+  const { errors } = props;
+  const nameError = errors.project?.message;
+  const tagError = errors.tag?.message;
+  let msg = null;
+
+  if (nameError == 'empty' && !tagError) {
+    msg = 'Project Name must not be empty.';
+  } else if (nameError == 'invalid' && !tagError) {
+    msg = "Project Name must contain only alphanumeric characters, '-', or '_'.";
+  } else if (nameError == 'empty' && tagError == 'invalid') {
+    msg = "Project Name must not be empty and Tag must contain only alphanumeric characters, '-', or '_'.";
+  } else if (nameError == 'invalid' && tagError == 'invalid') {
+    msg = "Project Name and Tag must contain only alphanumeric characters, '-', or '_'.";
+  } else if (nameError == 'empty' && tagError == 'empty') {
+    msg = 'Project Name and Tag must not be empty.';
+  } else if (nameError == 'invalid' && tagError == 'empty') {
+    msg = "Project Name must contain only alphanumeric characters, '-', or '_' and Tag must not be empty.";
+  } else if (!nameError && tagError == 'empty') {
+    msg = 'Project Tag must not be empty.';
+  } else if (!nameError && tagError == 'invalid') {
+    msg = "Project Tag must contain only alphanumeric characters, '-', or '_'.";
+  }
+
```

Comment:
I've seen this in a lot of places... wondering two things:

1. Can this be abstracted to a function, and
2. All these `if-else`'s seem like an anti-pattern. Can it not be a `switch-case`?

Reply:
I wanted to make it a function, but wasn't sure where to put it

Reply:
Yeah good q. I think I have a `utils.ts` file somewhere I put small little helpers like these

Reply:
yeah. `web/src/utils/[file].ts`